include/zos-char-util.h

```diff
@@ -200,73 +200,8 @@ class __Z_EXPORT __conv_off {
-  static int last_ccsid = 819;
-  static const unsigned char _tab_a[256] __attribute__((aligned(8))) = {
-      1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-      0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-  };
-  static const unsigned char _tab_e[256] __attribute__((aligned(8))) = {
-      1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
-      1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0,
-      0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1,
-      1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
-      1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0,
-      0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1,
-      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
-      0, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
-      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
-  };
-  unsigned long bytes;
-  unsigned long code_out;
-  const unsigned char *start;
-
-  bytes = max_len;
-  code_out = 0;
-  start = str;
-  __asm volatile(" trte %1,%3,0\n"
-                 " jo *-4\n"
-                 : NR("+",r3)(bytes), NR("+",r2)(str), "+r"(bytes), "+r"(code_out)
-                 : NR("",r1)(_tab_a)
-                 :);
-  unsigned a_len = str - start;
-
-  bytes = max_len;
-  code_out = 0;
-  str = start;
-  __asm volatile(" trte %1,%3,0\n"
-                 " jo *-4\n"
-                 : NR("+",r3)(bytes), NR("+",r2)(str), "+r"(bytes), "+r"(code_out)
-                 : NR("",r1)(_tab_e)
-                 :);
-  unsigned e_len = str - start;
-  if (a_len > e_len) {
-    *code_page = 819;
-    last_ccsid = 819;
-    *ambiguous = 0;
-    return a_len;
-  } else if (e_len > a_len) {
-    *code_page = 1047;
-    last_ccsid = 1047;
-    *ambiguous = 0;
-    return e_len;
-  }
-  *code_page = last_ccsid;
-  *ambiguous = 1;
-  return a_len;
-}
+__Z_EXPORT unsigned strlen_ae(const unsigned char *str, int *code_page,
```

Comment:
Probably a side question, but are user's of the library mean to be able to call the strlen_* functions or the rest of the functions in this header?  (should this function be exported or not?)

Reply:
Since it was inlined it's available and may be in use; I think it wouldn't hurt to export it. Foe the other functions in this header, Node.js uses some, e.g.  __guess_fd_ue, __set_autocvt_on_fd_stream, and an npm (node-report) uses class __auto_ascii.

Reply:
Thanks for the info.